torchgeo/trainers/instancesegmentation.py

```diff
@@ -0,0 +1,174 @@
+from typing import Any                                           
+import torch.nn as nn                                            
+import torch                                                   
+from torch import Tensor                                        
+from torchmetrics.detection.mean_ap import MeanAveragePrecision  
+from torchvision.models.detection import maskrcnn_resnet50_fpn   
+from ultralytics import YOLO  

```

Comment:
We would need to add a dependency on this which we don't currently have. I'm hesitant to do that given that they just had their PyPI account hacked and pushed out bitcoin miners for several releases. But they are quite well known, so it may be difficult not to support them. Let me ask around and see if others have opinions.